seqr/views/utils/terra_api_utils.py

```diff
@@ -192,10 +194,55 @@ def user_get_workspace_acl(user, workspace_namespace, workspace_name):
               "canCompute": true
             }
           }
+
     """
     path = "api/workspaces/{0}/{1}/acl".format(workspace_namespace, workspace_name)
     try:
         return _user_anvil_call('get', path, user).get('acl', {})
+    # Exceptions are handled to avoid error when a non-anvil user trying to validate permissions on AnVIL
     except (TerraNotFoundException, PermissionDenied) as et:
         logger.warning(str(et))
         return {}
+
+
+def update_workspace_acl(user, workspace_namespace, workspace_name, acl):
+    """
+    Update workspace acl.
+
+    The user must have the "can share" privilege for the workspace.
+
+    :param user: the seqr user object
+    :param workspace_namespace: namespace or billing project name of the workspace
+    :param workspace_name: name of the workspace on AnVIL. The name will also be used as the name of project in seqr
+    :param acl: the list of acl to be updated
+    :return: an object about the operation. See details at https://api.firecloud.org/#/Workspaces/updateWorkspaceACL
+    """
+    path = "api/workspaces/{0}/{1}/acl".format(workspace_namespace, workspace_name)
+    return _user_anvil_call('patch', path, user, data=acl)
+
+
+def add_service_account(user, workspace_namespace, workspace_name):
+    """
+    Add the seqr service account to the workspace on AnVIL.
+
+    The user must have the "can share" privilege for the workspace.
+
+    :param user: the seqr user object
+    :param workspace_namespace: namespace or billing project name of the workspace
+    :param workspace_name: name of the workspace on AnVIL. The name will also be used as the name of project in seqr
+    :return: Success: True, Fail: False
+    """
+    old_acl = user_get_workspace_acl(user, workspace_namespace, workspace_name)
+    service_account = old_acl.get(SERVICE_ACCOUNT_FOR_ANVIL)
+    if service_account and not service_account['pending']:
+        return True
+    acl = [
+             {
+               "email": SERVICE_ACCOUNT_FOR_ANVIL,
+               "accessLevel": "READER",
+               "canShare": False,
+               "canCompute": False
+             }
+          ]
+    users_updated = update_workspace_acl(user, workspace_namespace, workspace_name, json.dumps(acl))['usersUpdated']
+    return users_updated and users_updated[0]['email'] == SERVICE_ACCOUNT_FOR_ANVIL
```

Comment:
as mentioned above, I think it would be better raise exceptions if things fail rather than return true/false

Reply:
It seems that fewer efforts are required by checking the result in the calling function than throwing an exception here.

Reply:
I have figured out that adding an exception is better.